### PR TITLE
Move from CircleCI to GH Actions

### PR DESCRIPTION
**Pull request by cjyabraham**
This migrates our CI process from CircleCI to GH Actions. It also corrects several links in the docs and updates the README.

I tried to get the "Details" link in the PR build steps to link directly to the test instance, however, couldn't figure that one out. Maybe an idea for the future.

### CODE REVIEW

File: .github/workflows/build_deploy_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+          echo "export ARTIFACTS_DIR_URL='${GITHUB_WORKFLOW_URL}/#artifacts'" >> $BASH_ENV
+
+  build_php:
+    container:
+      image: quay.io/pantheon-public/build-tools-ci:6.x
+      options: --user root
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (thetwopct) [inline .github/workflows/build_deploy_and_test.yml:82]: Any particular reason why we are using this old version of checkout?

File: .github/workflows/build_deploy_and_test.yml [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+      - name: setup-environment-vars
+        run: /build-tools-ci/scripts/set-environment
+
+      - name: run PHP build step
+        run: ./.github/workflows/php
+
+  code_sniff_unit_test:
+    runs-on: ubuntu-latest
+    container:
+      image: drupaldocker/php:7.2-cli
```
Comment (thetwopct) [inline .github/workflows/build_deploy_and_test.yml:131]: Is it possible to use a PHP version thats more current?

**Review by thetwopct — APPROVED** [review]

File: .github/workflows/build_deploy_and_test.yml (hunk 1 above)
Comment (cjyabraham) [reply .github/workflows/build_deploy_and_test.yml]: fixed

File: .github/workflows/build_deploy_and_test.yml (hunk 2 above)
Comment (cjyabraham) [reply .github/workflows/build_deploy_and_test.yml]: I swapped in a different container

**Pull request merged**